torch/csrc/jit/python/pybind_utils.cpp

```diff
@@ -229,6 +233,9 @@ IValue toIValue(py::handle obj, const TypePtr& type, c10::optional<int32_t> N) {
         return py::cast<int64_t>(obj);
       } else if (py::isinstance<py::float_>(obj)) {
         return py::cast<double>(obj);
+      } else if (PyComplex_CheckExact(obj.ptr())) {
```

Comment:
Dumb question but why does this line not say something like `py::isinstance<py::complex_>(...)` like all the others?

Reply:
Complex type is not available in pybind (see doc here: https://pybind11.readthedocs.io/en/stable/advanced/pycpp/object.html#accessing-python-libraries-from-c) so I just pass the underlying `PyObject*` to `PyComplex_CheckExact` to check whether it's complex